File: sdkwork-backend-sdk-flutter/lib/src/api/image_generation.dart
import '../http/client.dart';
import '../models.dart';

class ImageGenerationApi {
  final HttpClient _client;
  
  ImageGenerationApi(this._client);

  /// Create image generation task
  Future<PlusApiResultGenerateImageVO?> create(GenerateImageForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/image/create'), body: body, contentType: 'application/json');
    return response is PlusApiResultGenerateImageVO ? response : null;
  }

  /// Get image generation result
  Future<PlusApiResultGenerateImageVO?> getResult(String taskId) async {
    final response = await _client.get(ApiPaths.backendPath('/generation/image/result/${taskId}'));
    return response is PlusApiResultGenerateImageVO ? response : null;
  }
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/image_generation.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import GenerateImageForm, PlusApiResultGenerateImageVO

class ImageGenerationApi:
    """image_generation API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create(self, body: GenerateImageForm) -> PlusApiResultGenerateImageVO:
        """Create image generation task"""
        return self._client.post(f"/backend/v3/api/generation/image/create", json=body)

    def get_result(self, taskId: str) -> PlusApiResultGenerateImageVO:
        """Get image generation result"""
        return self._client.get(f"/backend/v3/api/generation/image/result/{taskId}")
